src/main/java/za/co/neslotech/spring/trainspringbootinsight/auth/filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,96 @@
+package za.co.neslotech.spring.trainspringbootinsight.auth.filter;
+
+import io.micrometer.common.lang.NonNull;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import za.co.neslotech.spring.trainspringbootinsight.exception.InvalidAuthorizationRequest;
+import za.co.neslotech.spring.trainspringbootinsight.service.JwtAuthenticationService;
+
+import java.io.IOException;
+
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final String HEADER_AUTHORIZATION = "Authorization";
+    private static final String AUTHORIZATION_BEARER = "Bearer";
+
+    private final JwtAuthenticationService jwtService;
+    private final UserDetailsService userDetailsService;
+    private final HandlerExceptionResolver resolver;
+
+    public JwtAuthenticationFilter(final JwtAuthenticationService jwtService,
+                                   final UserDetailsService userDetailsService,
+                                   @Qualifier("handlerExceptionResolver") final HandlerExceptionResolver resolver) {
+        this.jwtService = jwtService;
+        this.userDetailsService = userDetailsService;
+        this.resolver = resolver;
+    }
+
+    @Override
+    protected void doFilterInternal(@NonNull final HttpServletRequest request,
+                                    @NonNull final HttpServletResponse response, @NonNull final FilterChain filterChain)
+    throws ServletException, IOException {
+
+        try {
+            String path = request.getRequestURI();
+            if (!path.startsWith("/api/v1/auth"))
+                processJwtToken(request);
```

Comment:
Small but we like the standard of keeping the curly braces present for if statements, just keeps the scope definition more defined and explicit.